### DISCUSSION TITLE
What to look for when fast gestures break animations?

### DISCUSSION THREAD

**Discussion by tomwaitforitmy** [software-mansion/react-native-gesture-handler#2054] [Q&A]
I have recently tried my first project with react-native-reanimated and react-native-gesture-handler. I wanted to implement a drag and sort (similar to this one [shown here](https://www.youtube.com/watch?v=Ia2y0GDEuPc)), but with elements of different heights. 

My code is now at a stage where everything works well when using gestures at moderate speed, however, if I move quickly, things break. It looks like some order steps are skipped:
![FailedAnimation](https://user-images.githubusercontent.com/25305424/168618966-c45ca862-2626-452e-800f-48e084ed6839.gif)

Since everything runs smoothly when gestures are performed at moderate speed, I have no clue what to check. Any help is highly appreciated. Thanks in advance.

You can find my component here:
Snack: https://snack.expo.dev/@tomwaitforitmy/movable-element-container
(only works on IOS for unknown reasons).
[github: components MovableElementContainer, MovableElementContainerConfig, MovableElementContainerUtil and MovableElement](https://github.com/tomwaitforitmy/mangi-bevi/blob/main/components/MovableElementContainer.js)

I have doubts if this topic belongs to react-native-gesture-handler or react-native-reanimated and therefore I will post on both projects until told otherwise. [Here you find the other topic](https://github.com/software-mansion/react-native-reanimated/discussions/3252).

**Comment by tomwaitforitmy** [comment]
Still looking for help. Any suggestions are very welcome.

**Comment by tomwaitforitmy** [comment]
Still looking for help. Any suggestions are very welcome.

**Reply by JasonCareter** [reply]
Consider of including the speed of pan gesture . After speed slow down and item position satisfied “exchanged” , then exchange . If the speed’s fast and order changed, just ignore it. don’t run animation until speed’s slow down

**Reply by tomwaitforitmy** [reply]
Alright, sounds like an idea, but wouldn't that possibly yield to items that never swap positions? Did you make good experience with this in similar things?

**Comment by tomwaitforitmy** [comment]
My current assumption is: If the gesture is performed so fast, that the sampling rate skips one or more of the elements, my algorithm messes up the positions. Is there a decent way to completely cancel fast gestures and reset all to start?
